subsys/zigbee/osif/zb_nrf_transceiver.c

```diff
@@ -23,21 +23,26 @@
 
 #if defined(NRF52840_XXAA) || defined(NRF52811_XXAA) || \
 defined(NRF5340_XXAA_NETWORK) || defined(NRF5340_XXAA_APPLICATION)
-/* dBm value corresponding to value 0 in the EDSAMPLE register. */
-#define ZBOSS_ED_MIN_DBM       (-92)
+/* Minimum value in dBm detectable by the radio. */
+#define MIN_RADIO_SENSITIVITY (-92)
 /* Factor needed to calculate the ED result based on the data from the RADIO peripheral. */
 #define ZBOSS_ED_RESULT_FACTOR 4
 
-#elif defined(NRF52833_XXAA) || defined(NRF52820_XXAA)
-/* dBm value corresponding to value 0 in the EDSAMPLE register. */
-#define ZBOSS_ED_MIN_DBM       (-93)
+#elif defined(NRF52833_XXAA) || defined(NRF52820_XXAA) \
+|| defined(NRF5340_XXAA_NETWORK) || defined(NRF5340_XXAA_APPLICATION)
+/* Minimum value in dBm detectable by the radio. */
+#define MIN_RADIO_SENSITIVITY (-93)
 /* Factor needed to calculate the ED result based on the data from the RADIO peripheral. */
 #define ZBOSS_ED_RESULT_FACTOR 5
 
 #else
 #error "Selected chip is not supported."
 #endif
 
+/* dBm value corresponding to value 0 of the energy scan result. */
+#define ZBOSS_ED_MIN_DBM (-75)
+/* dBm value corresponding to value 255 of the energy scan result. */
+#define ZBOSS_ED_MAX_DBM (MIN_RADIO_SENSITIVITY + (255/ZBOSS_ED_RESULT_FACTOR))
```

Comment:
If we are 100% sure that the Radio Driver simply gets the floor of the calculated dBm value, than we can keep it. If it does some more interesting adjustments, it will be safer to write `(MIN_RADIO_SENSITIVITY + (255/ZBOSS_ED_RESULT_FACTOR) + 1)`

Reply:
The radio driver uses exactly the same formulas as the zb_nrf_transceiver, so there are no complicated adjustments here.